python/ray/serve/backend_worker.py

```diff
@@ -256,7 +242,28 @@ async def invoke_single(self, request_item: Query) -> Any:
 
         start = time.time()
         try:
-            result = await method_to_call(arg)
+            if self.config.internal_metadata.is_asgi_app:
+                request: Request = arg
+                scope = request.scope
+                root_path = self.config.internal_metadata.path_prefix
+
+                # The incoming scope["path"] contains prefixed path and it
+                # won't be stripped by FastAPI.
+                request.scope["path"] = scope["path"].replace(root_path, "", 1)
+                # root_path is used such that the reverse look up and
+                # redirection works.
+                request.scope["root_path"] = root_path
```

Comment:
Hmm why won't this be handled by fastapi? It still seems like we shouldn't need to be mucking around with the fields directly

Reply:
root_path only works for the redirects, it actually is not used by routing. FastAPI expects the incoming route is cleaned and root_path is stripped by proxy already. We are essentially performing the path stripping here.

Reply:
Ok, cool, makes sense